File: backends/kotlin-ktor/src/main/kotlin/com/authorization/infrastructure/cache/RedisClient.kt
/*
 * Redis クライアント生成モジュール。
 *
 * @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
 */
package com.authorization.infrastructure.cache

import com.authorization.config.Config
import redis.clients.jedis.JedisPool
import redis.clients.jedis.JedisPoolConfig

/**
 * 設定をもとに Redis 接続プールを生成します。
 *
 * @param cfg アプリケーション設定
 * @return Redis 接続プール
 */
fun newRedisPool(cfg: Config): JedisPool {
    val poolConfig = JedisPoolConfig().apply {
        maxTotal = 16
    }
    return if (cfg.redis.password.isBlank()) {
        JedisPool(poolConfig, cfg.redis.host, cfg.redis.port, 2000, null, cfg.redis.db)
    } else {
        JedisPool(poolConfig, cfg.redis.host, cfg.redis.port, 2000, cfg.redis.password, cfg.redis.db)
    }
}

File: backends/ruby-hanami/app/infrastructure/cache/redis_client.rb
# frozen_string_literal: true
#
# Redis クライアントファクトリを定義するモジュール。
#
# @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>

module Infrastructure
  module Cache
    # Redis クライアントを生成するファクトリクラスです。
    # @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
    class RedisClient
      # @param cfg [ConfigLoader] 設定
      # @return [Redis] Redis クライアント
      def self.new_client(cfg = ConfigLoader.load)
        Redis.new(
          host:     cfg.redis.host,
          port:     cfg.redis.port,
          password: cfg.redis.password.empty? ? nil : cfg.redis.password,
          db:       cfg.redis.db,
        )
      end
    end
  end
end
